### DISCUSSION TITLE
Replace opam dependency with ocamlPackages?

### DISCUSSION THREAD

**Discussion by mistzzt** [tweag/opam-nix#84] [Q&A] [answered]
Is it possible to replace some dependencies (e.g., z3 ocaml binding) defined in `.opam` files with ones in nixPkgs.ocamlPackages?

I have some compilation errors with opam-nix generated opam packages, so I wonder if I can just replace them with ones in `nixPkgs.ocamlPackages`. I can send a minimal reproduction example if needed.

**Comment by balsoft** [comment, marked as answer]
In general, no. The dependency tree for every package needs to be conistent, i.e. have exactly the same versions of every dependency, including ocaml itself. Unless you are extremely lucky, or force that to be the case, it will break. (And even otherwise, there will probably be some other issues that pop up).

The best solution is to fix problems with the packages (and then contribute those fixes to opam-nix :wink: )
